File: apps/error-tracking-upload-source-maps/react-native-expo/modules/native-crash-test/ios/NativeCrashTestModule.swift
import ExpoModulesCore

public class NativeCrashTestModule: Module {
  public func definition() -> ModuleDefinition {
    Name("NativeCrashTest")

    // fatalError aborts the process with SIGABRT, which PostHog's native
    // crash autocapture records via its signal handler. The report is
    // written to disk and uploaded on the next app launch.
    Function("crash") {
      fatalError("PostHog native crash test (iOS)")
    }
  }
}

File: apps/error-tracking-upload-source-maps/react-native-expo/modules/native-crash-test/android/src/main/java/expo/modules/nativecrashtest/NativeCrashTestModule.kt
package expo.modules.nativecrashtest

import android.os.Handler
import android.os.Looper
import expo.modules.kotlin.modules.Module
import expo.modules.kotlin.modules.ModuleDefinition

class NativeCrashTestModule : Module() {
  override fun definition() = ModuleDefinition {
    Name("NativeCrashTest")

    // Posts to the main thread so the exception escapes React Native's
    // native-module call guards and crashes the app as a real uncaught
    // JVM exception (what PostHog's native crash autocapture records).
    Function("crash") {
      Handler(Looper.getMainLooper()).post {
        throw RuntimeException("PostHog native crash test (Android)")
      }
    }
  }
}
